### PR TITLE
Simpified `_check_distribution`.

### PR DESCRIPTION
**Pull request by PhilipMay**
PR as requested here: https://github.com/optuna/optuna/pull/908#issuecomment-586818909

### CODE REVIEW

File: optuna/trial.py [hunk 1]
```diff
@@ -554,35 +554,19 @@ def _is_relative_param(self, name, distribution):
...
+        if old_distribution != distribution:
             warnings.warn('Inconsistent parameter values for distribution with name "{}"! '
                           'This might be a configuration mistake. '
                           'Optuna allows to call the same distribution with the same '
                           'name more then once in a trial. '
                           'When the parameter values are inconsistent optuna only '
                           'uses the values of the first call and ignores all following. '
                           'Using these values: {}'
-                          .format(name, old_distribution_in_trial_values), RuntimeWarning)
+                          .format(name, str(old_distribution)), RuntimeWarning)
```
Comment (sile) [inline optuna/trial.py]: I'm not sure that it's good to customize `__str__` implemenation for the specific purpose (and I think that there is a room to discuss what is the best string representation of distribution class).
How about using dict representation instead (as in the previous message)? 
```suggestion
                          .format(name, old_distribution._asdict()), RuntimeWarning)
```

Comment (PhilipMay) [reply optuna/trial.py]: made this change

**Comment by PhilipMay** [comment]
@sile what is the problem with doctest here?

**Comment by sile** [comment]
> what is the problem with doctest here?

I'm not sure but it might be related to #942.

File: optuna/distributions.py [hunk 2]
```diff
@@ -143,6 +143,14 @@ def _contains(self, param_value_in_internal_repr):
         else:
             return self.low <= value and value < self.high
 
+    def _asdict(self):
```
Comment (sile) [inline optuna/distributions.py]: Sorry for the lack of explanation. `BaseDistribution` already have [the default `_asdict` implementation](https://github.com/optuna/optuna/blob/4ad45e8086b09f9e8733dca90f463d433ef10288/optuna/distributions.py#L83), so you don't need to implement it for each distribution class.

Comment (PhilipMay) [reply optuna/distributions.py]: oops. Fixed it.

**Comment by PhilipMay** [comment]
> > what is the problem with doctest here?
> 
> I'm not sure but it might be related to #942. Merging master branch might resolve the problem.

done

**Comment by sile** [comment]
Thanks!

**Comment by sile** [comment]
The CI failed again because of another problem (new XGBGoost release)... 
The problem was fixed by https://github.com/optuna/optuna/pull/948, please re-merge master branch.

File: tests/test_distributions.py [hunk 3]
```diff
@@ -235,3 +235,27 @@ def test_repr():
 
     for d in EXAMPLE_DISTRIBUTIONS.values():
         assert d == eval('distributions.' + repr(d))
+
+
+def test_str_uniform_distribution():
```
Comment (sile) [inline tests/test_distributions.py]: Please update test names.
```suggestion
def test_uniform_distribution_as_dict():
```

**Comment by hvy** [comment]
Just FYI, I tested retriggering the CU build.

File: tests/test_distributions.py (hunk 3 above)
Comment (PhilipMay) [reply tests/test_distributions.py]: I renamed to `_asdict`.

**Comment by PhilipMay** [comment]
Everything should be fine now.

**Comment by sile** [comment]
Hmm, the code seems fine, but unnecessary diffs are being appeared... 
https://github.com/optuna/optuna/pull/937/files

Could you try executing the following commands to suppress the extra diffs?
```console
$ git reset --hard da5e88870f17e8cf5571ffd30e8a582c54d08fe2
$ git cherry-pick 28a4ccbc43f182f545748cd2d880f413f51e9113
$ git push -f simple_check_distribution
```

**Comment by PhilipMay** [comment]
> Hmm, the code seems fine, but unnecessary diffs are being appeared...
> https://github.com/optuna/optuna/pull/937/files
> 
> Could you try executing the following commands to suppress the extra diffs?
> 
> ```
> $ git reset --hard da5e88870f17e8cf5571ffd30e8a582c54d08fe2
> $ git cherry-pick 28a4ccbc43f182f545748cd2d880f413f51e9113
> $ git push -f simple_check_distribution
> ```

I have done this:
``` bash
git reset --hard da5e88870f17e8cf5571ffd30e8a582c54d08fe2
git cherry-pick 28a4ccbc43f182f545748cd2d880f413f51e9113
git push -f 
```

**Pull request merged**